src/js/components/account/change-password.js

```diff
@@ -0,0 +1,90 @@
+import { API } from "../../api";
+import {
+  getDiv,
+  getForm,
+  getInput,
+  getLabel,
+  getSubmitButton,
+} from "../../utils/element-factory";
+
+export class ChangePassword {
+  constructor() {
+    const $successTemplate = $(".styles .form-styles .w-form-done").clone();
+    const $failTemplate = $(".styles .form-styles .w-form-fail").clone();
+
+    const defaultBaseUrl = "https://muni-portal-backend.openup.org.za";
+    const endPoint = "/api/accounts/change-password/";
+
+    const $changeFormContainer = getDiv("form w-form");
+    const $form = getForm(`${defaultBaseUrl}${endPoint}`, "post");
+    const $submitButton = getSubmitButton("Change Password");
+
+    const $formElementsContainer = $("<div />");
+    $formElementsContainer.append(getLabel("current password"));
+    $formElementsContainer.append(getInput("password", "old_password"));
+
+    $formElementsContainer.append(getLabel("new password"));
+    $formElementsContainer.append(getInput("password", "password"));
+
+    $formElementsContainer.append(getLabel("confirm new password"));
+    $formElementsContainer.append(getInput("password", "password_confirm"));
+
+    $form.append($formElementsContainer);
+    $form.append($submitButton);
+
+    $changeFormContainer.append($successTemplate);
+    $changeFormContainer.append($failTemplate);
+    $changeFormContainer.append($form);
+
+    $form.submit((event) => {
+      event.preventDefault();
+      this.changePassword(endPoint, $form, $successTemplate, $failTemplate);
+    });
+
+    this.$element = $changeFormContainer;
+  }
+
+  /**
+   * Calls API end point to change user's password
+   * @param {String} endPoint - the API endpoint
+   * @param {jqObject} $form - the form to submit
+   * @param {jqObject} $success - reference to the success template
+   * @param {jqObject} $fail  - reference to the failure template
+   */
+  changePassword(endPoint, $form, $success, $fail) {
+    const api = new API();
+    const response = api.changePassword(endPoint, $form.serialize());
+    response
+      .done((response, textStatus) => {
+        if (textStatus === "success") {
+          $form.hide();
+          $success
+            .empty()
+            .append("Your password has been changed successfully.")
+            .show();
```

Comment:
There are other places where we need to fix this as well but, for these messages, we should:

1. Ensure that the containing element has `tabindex="-1"` if it is not natively focusable
2. Call `.focus()` on the containing element

This is to ensure that a screen reader will read the message or else a blind person will have no idea the action was successful.

Reply:
Thanks @schalkneethling - do you have an example of where we currently do this? Or would this be the first time we do it? I just want to make sure to stick to some kind of pattern.

Reply:
I've pushed some commits that set `focus()` and `attr('tabindex', -1)` on the messages. Let me know if this is correct or if there's a better way to do this to avoid repeating the same pattern